modules/seqexec/server/src/main/scala/seqexec/server/altair/AltairController.scala

```diff
@@ -3,26 +3,29 @@
 
 package seqexec.server.altair
 
-import seqexec.server.tcs.Gaos.Reason
+import seqexec.server.tcs.Gaos.{PauseReason, ResumeReason}
 import squants.Time
 
 trait AltairController[F[_]] {
   import AltairController._
 
-  def pause(reasons: Set[Reason]): F[Unit]
-  def resume(reasons: Set[Reason], config: AltairConfig): F[Unit]
-  def observe(expTime: Time): F[Unit]
-  def endObserve: F[Unit]
+  def pause(reasons: Set[PauseReason], fieldLens: FieldLens)(cfg: AltairConfig): F[Unit]
+  def resume(reasons: Set[ResumeReason])(cfg: AltairConfig): F[Unit]
+  def observe(expTime: Time)(cfg: AltairConfig): F[Unit]
+  def endObserve(cfg: AltairConfig): F[Unit]
+  def config(cfg: AltairConfig): F[Unit]
```

Comment:
Since all these methods now take `AltairConfig` could you just pass it when you create the instance? Your `F` could be `Kleisli[IO, AltairConfig, ?]` for example.

Reply:
Maybe, but Altair is not complete, and details may change. I will revisit this after it is done.